### PR TITLE
fix debug bar cm6 compat

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Fix debug bar compatibility with CodeMirror 6.

### CODE REVIEW

File: lib/bundles/codemirror.js [hunk 1]
```diff
@@ -34,12 +34,29 @@
 // Codemirror base lib
 
 import {EditorView, basicSetup} from "codemirror";
-import {css} from "@codemirror/lang-css"
+import {css} from "@codemirror/lang-css";
+import {json} from "@codemirror/lang-json";
+import {sql} from "@codemirror/lang-sql";
+import {highlightTree} from "@lezer/highlight";
+import {defaultHighlightStyle} from "@codemirror/language";
```
Comment (cedric-anne) [inline lib/bundles/codemirror.js:41]: `@codemirror/language` should be added to our dependencies if we use in our code.

File: lib/bundles/codemirror.js [hunk 2]
```diff
@@ -34,12 +34,29 @@
...
 window.CodeMirror = {
     EditorView: EditorView,
     setup:      basicSetup,
     languages:  {
         css: css,
+        json: json,
+        sql: sql,
     },
+    defaultHighlightStyle: defaultHighlightStyle,
+    runMode: (textContent, language, callback) => {
```
Comment (cedric-anne) [inline lib/bundles/codemirror.js:52]: Seems to be inspired by https://discuss.codemirror.net/t/static-highlighting-using-cm-v6/3420

```suggestion
    runMode: (textContent, language, callback) => {
        // see https://discuss.codemirror.net/t/static-highlighting-using-cm-v6/3420
```

File: js/modules/Debug/Debug.js [hunk 3]
```diff
@@ -576,14 +584,15 @@ window.GLPI.Debug = new class Debug {
...
+            const editor = new window.CodeMirror.EditorView({
+                extensions: [
+                    window.CodeMirror.setup,
+                    window.CodeMirror.languages.json(),
+                    window.CodeMirror.EditorView.lineWrapping
+                ],
+                doc: data_string
             });
+            container.append(editor.dom);
             container.data('editor', editor);
```
Comment (cedric-anne) [inline js/modules/Debug/Debug.js:596]: Is this still used?

**Review by cedric-anne — COMMENTED** [review]
Seems to be OK, except that array/objects are not folded by default in globals variables tabs.

**Comment by cconard96** [comment]
> Seems to be OK, except that array/objects are not folded by default in globals variables tabs.

Is there any better documentation for this library beyond the official mess on their site?

**Comment by cedric-anne** [comment]
> > Seems to be OK, except that array/objects are not folded by default in globals variables tabs.
> 
> Is there any better documentation for this library beyond the official mess on their site?

Not really. We can still merge this PR. The only place where it can be a pain to have unfolded objects/arrays is the `$_SESSION` tab, but it is probably one of the less used.

**Review by cedric-anne — APPROVED** [review]

**Comment by cconard96** [comment]
Manual folding doesn't even work properly anyways. Their parser just gives up if there are a lot of lines so it only really does what is showing. When sections go off the screen, it doesn't show it as foldable until you scroll to load it all.

File: js/modules/Debug/Debug.js [hunk 4]
```diff
@@ -576,14 +584,15 @@ window.GLPI.Debug = new class Debug {
...
-                value: data_string,
-                mode: 'application/json',
-                lineNumbers: true,
-                readOnly: true,
-                foldGutter: true,
-                gutters: ['CodeMirror-linenumbers', 'CodeMirror-foldgutter'],
+            const editor = new window.CodeMirror.EditorView({
+                extensions: [
+                    window.CodeMirror.setup,
+                    window.CodeMirror.languages.json(),
```
Comment (cedric-anne) [inline js/modules/Debug/Debug.js:590]: This should make it readonly.
```suggestion
                    window.CodeMirror.languages.json(),
                    window.CodeMirror.EditorView.contentAttributes.of({contenteditable: false}),
```

**Pull request merged**